### PR TITLE
Install infection

### PR DESCRIPTION
**Pull request by Chemaclass**
## 📚 Description

Simila as we did on container and router, we want to install and get the benefits from infection php mutation testing framework on the core repository of gacela.

## 🔖 Changes

- install infection

## 🖼️  Status quo

Screenshot 2023-05-03 at 12 04 14

### CODE REVIEW

File: .github/workflows/mutation-testing.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+      - name: Get composer cache directory
+        id: composer-cache
+        run: echo "dir=$(composer config cache-files-dir)" >> $GITHUB_OUTPUT
+
+      - name: Cache dependencies
+        uses: actions/cache@v3.3.1
+        with:
+          path: ${{ steps.composer-cache.outputs.dir }}
+          key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
```
Comment (localheinz) [inline .github/workflows/mutation-testing.yml:33]: @Chemaclass 

`composer.lock` is currently not checked in and does not exist, so `actions/cache` can not assemble a cache key based on `composer.lock`.

Perhaps change this to

```suggestion
          key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.json') }}
```

or check in `composer.lock` - also helps with reproducible builds.

What do you think?

Comment (Chemaclass) [reply .github/workflows/mutation-testing.yml]: You're right. However, we are using the same line in the other workflows ([code-style](https://github.com/gacela-project/gacela/blob/main/.github/workflows/code-style.yml#L33), for example), and no issues there

**Comment by Chemaclass** [comment]
I am using the same as in the docs https://github.com/shivammathur/setup-php#cache-composer-dependencies but still something seems to be wrong... any more ideas, @localheinz ?

**Comment by Chemaclass** [comment]
I will merge this as it adds infection to the project. The problem with "composer caching" is another thing that we can postpone to another time.

**Pull request merged**